coap-core/src/main/java/com/mbed/coap/server/internal/CoapRequestId.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2011-2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mbed.coap.server.internal;
+
+import java.net.InetSocketAddress;
+import java.util.Objects;
+
+public class CoapRequestId implements Timestamped {
+
+    private final int mid;
+    private final InetSocketAddress sourceAddress;
+    private final transient long createdTimestampMillis;
+
+    public CoapRequestId(int mid, InetSocketAddress sourceAddress) {
+        this.mid = mid;
+        this.sourceAddress = sourceAddress;
+        this.createdTimestampMillis = System.currentTimeMillis();
```

Comment:
I'm thinking is it good createdTimestampMillis is needed here.

Reply:
It would be better if the implementation of the cache could do it without having this in the key.

Reply:
But I don't know if we want to change things too much at this point? It could be made better through iteration.